### PR TITLE
Upload artifacts as output of check-plugins check for debugging

### LINKED ISSUE #10116: Upload artifacts as output of check-plugins check for debugging
# Description

As step to ease the debugging process (https://github.com/rancher/dashboard/issues/10112), upload generated files on fail.

E.g.:

```yaml
    - name: Upload files
      uses: actions/upload-artifact@v3
      if: failure()
      with:
        name: test
        path: |
          /tmp/tmp.**/test-app
          !/tmp/tmp.**/test-app/node_modules/
          /tmp/tmp.**/pkg/test-pkg/
          !/tmp/tmp.**/pkg/test-pkg/node_modules/
        
```

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #10116

### Occurred changes and/or fixed issues

Added artifacts upload on failure with 2 days of retention for the paths:
```
          /tmp/tmp.**/test-app
          !/tmp/tmp.**/test-app/node_modules/
          !/tmp/tmp.**/test-app/pkg/test-pkg/node_modules/
```

### Technical notes summary

- ~Commented removal logic for the process.~ Conditionally persisted build, default to remove
- Added logs for removed paths
- Updated RegEx to include `-alpha.x`, `-releasex.x.x` and `-rc.x` while selecting versions
- Excluded paths for `elemental-ui/` and `kubewarden-ui/` for eslint

### CODE REVIEW

**Review by aalves08 — DISMISSED** [review]
I don't see any problems with it. lgtm. @nwmac any input here?

**Review by nwmac — DISMISSED** [review]
Like the idea - one ask on the folder removes, so that it still works cleanly when run locally for dev.

File: shell/scripts/test-plugins-build.sh [hunk 1]
```diff
@@ -73,7 +73,7 @@ export YARN_REGISTRY=$VERDACCIO_YARN_REGISTRY
 export NUXT_TELEMETRY_DISABLED=1
 
 # Remove test package from previous run, if present
-rm -rf ${BASE_DIR}/pkg/test-pkg
+# rm -rf ${BASE_DIR}/pkg/test-pkg
```
[the remark this answers is not in the document]
Comment (cnotv) [reply shell/scripts/test-plugins-build.sh]: I've also added a log with the path, so it's easier to see if it runs and what it removes.

**Comment by cnotv** [comment]
@aalves08 Overnight this error started to appear:

```
npm ERR! code EPUBLISHCONFLICT
npm ERR! publish fail Cannot publish over existing version.
npm ERR! publish fail Update the 'version' field in package.json and try again.
npm ERR! publish fail 
npm ERR! publish fail To automatically increment version numbers, see:
npm ERR! publish fail     npm help version
```

Any hint?

**Comment by cnotv** [comment]
> @aalves08 Overnight this error started to appear:
> 
> ```
> npm ERR! code EPUBLISHCONFLICT
> npm ERR! publish fail Cannot publish over existing version.
> npm ERR! publish fail Update the 'version' field in package.json and try again.
> npm ERR! publish fail 
> npm ERR! publish fail To automatically increment version numbers, see:
> npm ERR! publish fail     npm help version
> ```
> 
> Any hint?

@aalves08 @nwmac the issue seems to appear since the publication in NPM just to let you know https://www.npmjs.com/package/@rancher/components

**Comment by cnotv** [comment]
@aalves08 Solved 😄

**Review by aalves08 — DISMISSED** [review]
LGTM

**Comment by cnotv** [comment]
@aalves08 we need @nwmac approval due change file request

File: shell/scripts/test-plugins-build.sh [hunk 2]
```diff
@@ -130,23 +134,31 @@ if [ "${SKIP_STANDALONE}" == "false" ]; then
   echo "Cleaning temporary dir"
   popd > /dev/null
 
-  rm -rf ${DIR}
+  if [ $TEST_PERSIST_BUILD != "true" ]; then
+  echo "Removing folder ${DIR}"
```
Comment (richard-cox) [inline shell/scripts/test-plugins-build.sh:138]: indentation

File: shell/scripts/test-plugins-build.sh [hunk 3]
```diff
@@ -194,7 +209,10 @@ function clone_repo_test_extension_build() {
   popd
 
   # delete folder
-  rm -rf ${BASE_DIR}/$REPO_NAME
+  if [ $TEST_PERSIST_BUILD != "true" ]; then
+  echo "Removing folder ${BASE_DIR}/$REPO_NAME"
```
Comment (richard-cox) [inline shell/scripts/test-plugins-build.sh:213]: indentation

File: shell/scripts/test-plugins-build.sh [hunk 4]
```diff
@@ -73,17 +73,20 @@ export YARN_REGISTRY=$VERDACCIO_YARN_REGISTRY
...
+if [ $TEST_PERSIST_BUILD != "true" ]; then
+  echo "Removing folder ${BASE_DIR}/pkg/test-pkg"
+  rm -rf ${BASE_DIR}/pkg/test-pkg
+fi
 
 # We need to patch the version number of the shell, otherwise if we are running
 # with the currently published version, things will fail as those versions
 # are already published and Verdaccio will check, since it is a read-through cache
-sed -i.bak -e "s/\"version\": \"[0-9]*.[0-9]*.[0-9]*\",/\"version\": \"${SHELL_VERSION}\",/g" ${SHELL_DIR}/package.json
+sed -i.bak -e "s/\"version\": \"[0-9]*.[0-9]*.[0-9]*\(-alpha\.[0-9]*\|-release[0-9]*.[0-9]*.[0-9]*\|-rc\.[0-9]*\)\{0,1\}\",/\"version\": \"${SHELL_VERSION}\",/g" ${SHELL_DIR}/package.json
```
Comment (richard-cox) [inline shell/scripts/test-plugins-build.sh:84]: comment only. at some point we should just use the suggested semver regex https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string. if this is tested and works though, no need to go through another round

**Review by richard-cox — CHANGES_REQUESTED** [review]
three minor comments about indentation.

i haven't tested the `sed` reg ex, but tested the non-escaped reg ex for semver and look ok

**Review by richard-cox — APPROVED** [review]

**Pull request merged**